azure-kusto-data/azure/kusto/data/security.py

```diff
@@ -1,254 +1,71 @@
-from azure.core.credentials import AccessToken
-
-from .exceptions import KustoClientError, KustoAuthenticationError
-
-
-def _get_azure_cli_auth_token() -> dict:
-    """
-    Try to get the az cli authenticated token
-    :return: refresh token
-    """
-    import os
-
-    try:
-        # this makes it cleaner, but in case azure cli is not present on virtual env,
-        # but cli exists on computer, we can try and manually get the token from the cache
-        from azure.cli.core._profile import Profile
-        from azure.cli.core._session import ACCOUNT
-        from azure.cli.core._environment import get_config_dir
-
-        azure_folder = get_config_dir()
-        ACCOUNT.load(os.path.join(azure_folder, "azureProfile.json"))
-        profile = Profile(storage=ACCOUNT)
-        token_data = profile.get_raw_token()[0][2]
-
-        return token_data
-
-    except ModuleNotFoundError:
-        try:
-            import os
-            import json
-
-            folder = os.getenv("AZURE_CONFIG_DIR", None) or os.path.expanduser(os.path.join("~", ".azure"))
-            token_path = os.path.join(folder, "accessTokens.json")
-            with open(token_path) as f:
-                data = json.load(f)
-
-            # TODO: not sure I should take the first
-            return data[0]
-        except Exception as e:
-            raise KustoClientError("Azure cli token was not found. Please run 'az login' to setup account.", e)
-
-
-@unique
-class AuthenticationMethod(Enum):
-    """Enum representing all authentication methods available in Kusto with Python."""
-
-    aad_username_password = "aad_username_password"
-    aad_application_key = "aad_application_key"
-    aad_application_certificate = "aad_application_certificate"
-    aad_application_certificate_sni = "aad_application_certificate_sni"
-    aad_device_login = "aad_device_login"
-    aad_token = "aad_token"
-    managed_service_identity = "managed_service_identity"
-    az_cli_profile = "az_cli_profile"
-    token_provider = "token_provider"
-
-
-CLOUD_LOGIN_URL = "https://login.microsoftonline.com/"
+from .exceptions import KustoAuthenticationError
+from ._token_providers import *
 
 
 class _AadHelper:
-    authentication_method = None
-    auth_context = None
-    msi_auth_context = None
-    username = None
     kusto_uri = None
     authority_uri = None
-    client_id = None
-    password = None
-    thumbprint = None
-    private_certificate = None
-    public_certificate = None
-    msi_params = None
     token_provider = None
 
     def __init__(self, kcsb: "KustoConnectionStringBuilder"):
         self.kusto_uri = "{0.scheme}://{0.hostname}".format(urlparse(kcsb.data_source))
         self.username = None
 
+        cloud_info = CloudSettings.get_cloud_info()
+        if cloud_info is None:
+            raise KustoClientError("Unable to detect cloud instance from DNS Suffix of Kusto Connection String [" + self.kusto_uri + "]")
```

Comment:
The message implies of the previous implementation.
1. Please edit it, preferably with go-to-action instruction.
2. Consider throw it from cloud info instead so when moving to the metadata endpoint we'll edit it there.